File: HandOnML/P05_AssociationRuleLearning/01_Apriori/apriori.py
# Apriori

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Data Preprocessing
dataset = pd.read_csv('Market_Basket_Optimisation.csv', header = None)
transactions = []
for i in range(0, 7501): # exclude uper bound 7501
    transactions.append([str(dataset.values[i,j]) for j in range(0, 20)]) # exclude index = 20

# Training Apriori on the dataset
from apyori import apriori
# get items were bought at least 3 times a day
rules = apriori(transactions, min_support = 3*7/7500, min_confidence = 0.2, min_lift = 3, min_length = 2)

# Visualising the results
results = list(rules)

File: HandOnML/P05_AssociationRuleLearning/01_Apriori/apriori.R
# Title     : Apriori
# Created by: trankhai
# Created on: 10/6/17

# Data preprocessing
# install.packages('arules')
library(arules)
dataset = read.csv('Market_Basket_Optimisation.csv', header = FALSE)
dataset = read.transactions('Market_Basket_Optimisation.csv', sep = ',', rm.duplicates =TRUE)
summary(dataset)
itemFrequencyPlot(dataset, topN = 20)

# Training Apriori on the dataset
# Support: Buy 3 item for 7 day on week per total 7500 transactions
rules = apriori(data = dataset, parameter = list(support = 3*7/7500, confidence = 0.4))

# Visualising the results
inspect(sort(rules, by = 'lift')[1:10])